### PR TITLE
refactor: fix PHPStan errors

### PR DESCRIPTION
**Pull request by pawelkg**
**Description**
Fix errors from phpstan-baseline.php

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [ ] Conforms to style guide

### CODE REVIEW

**Comment by samsonasik** [comment]
please create multiple PRs for per-directory basis to make easier to review, also, better fix per error identifier instead of try to fix everything, then you can rebase when one of PRs merged.

**Comment by kenjis** [comment]
Sorry, we cannot review too big PRs like this. Your one commit is too big and having a lot of different changes,
and the commit messages do not make sense like "fix another PHPUnit DatabaseLive errors part2".
Commit messages should show 

> Now we don't have detailed rules on commits and its messages. But atomic commit is recommended. Keep your commits atomic. One commit for one change.

Read carefully: https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/workflow.md#committing

File: system/Config/AutoloadConfig.php [hunk 1]
```diff
@@ -77,6 +77,19 @@ class AutoloadConfig
...
+     * Helpers
+     * -------------------------------------------------------------------
+     * Prototype:
+     *   $helpers = [
+     *       'form',
+     *   ];
+     *
+     * @var list<string>
+     */
+    public $helpers = [];
```
Comment (kenjis) [inline system/Config/AutoloadConfig.php:91]: Why did you add the property?

File: system/Autoloader/Autoloader.php [hunk 2]
```diff
@@ -380,7 +383,13 @@ private function loadComposerNamespaces(ClassLoader $composer, array $composerPa
...
         }
 
-        if (! method_exists(InstalledVersions::class, 'getAllRawData')) {
+        preg_match(
+            '/\s*"plugin-api-version": "(?\'version\'\d+\.\d+\.\d+)"/m',
+            file_get_contents(__DIR__ . '/../../composer.lock'),
+            $matches
+        );
+
+        if (version_compare($matches['version'], '2.0.14', '<')) {
```
Comment (kenjis) [inline system/Autoloader/Autoloader.php:392]: Why did you change the code?

File: system/Autoloader/FileLocator.php [hunk 3]
```diff
@@ -145,10 +145,10 @@ public function getClassname(string $file): string
 
             if ((isset($tokens[$i - 2][1]) && ($tokens[$i - 2][1] === 'phpnamespace' || $tokens[$i - 2][1] === 'namespace')) || ($dlm && $tokens[$i - 1][0] === T_NS_SEPARATOR && $token[0] === T_STRING)) {
                 if (! $dlm) {
-                    $namespace = 0;
+                    $namespace = '';
```
Comment (kenjis) [inline system/Autoloader/FileLocator.php:148]: I don't know why the original code is `$namespace = 0;`.
If `$namespace` is set to `0`, is the case is just a bug?

Please make another PR to fix this issue.

**Comment by pawelkg** [comment]
Ok. I will prepare separate PRs with smaller, internally related scopes.

File: system/Autoloader/Autoloader.php (hunk 2 above)
Comment (pawelkg) [reply system/Autoloader/Autoloader.php:392]: This fix phpstan error:
`Call to function method_exists() with 'Composer\InstalledVersions' and 'getAllRawData' will always evaluate to true.`
But I think, that better way will be to remove this check and check composer version during `composer install` and `composer update`...

File: system/Autoloader/FileLocator.php (hunk 3 above)
Comment (pawelkg) [reply system/Autoloader/FileLocator.php:148]: After changing 151 line in this file it's failing:
1) CodeIgniter\Autoloader\FileLocatorCachedTest::testGetClassNameFromClassFile
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'CodeIgniter\Autoloader\FileLocatorTest'
+'0\CodeIgniter\Autoloader\FileLocatorTest'

File: system/Autoloader/Autoloader.php (hunk 2 above)
Comment (kenjis) [reply system/Autoloader/Autoloader.php:392]: Okay, I got why. But your code seems to take time more.
The PHPStan error is inevitable. Because we use the latest `composer` command.
So I think the error should be suppressed by `@phpstan-ignore-next-line` or `@phpstan-ignore-line`.
